### ISSUE TITLE
CLI overrides not updating in resolved plugin configuration

### ISSUE THREAD

**Issue by mattrlong** [badeball/cypress-cucumber-preprocessor#951]
### Current behavior

When attempting to override the plugin configuration from the CLI, the cypress environment variables are not appearing in the resolved plugin configuration. 

For example: 
```
✗ DEBUG=cypress:electron,cypress-configuration,cypress-cucumber-preprocessor npx cypress run --env filterSpecs=false
  cypress-cucumber-preprocessor resolved configuration { filterSpecs: true }
```

### Desired behavior

Updates to configuration through the CLI update plugin configuration as expected. 

### Test code to reproduce

configuration initially defined in .cypress-cucumber-preprocessorrc.json as follows: 

```
{
    "filterSpecs": true
}
```
### Versions

* **Cypress version**: 12.0.0
* **Preprocessor version**: 15.1.4
* **Node version**: 16.17.1

### Checklist

- [x] I've read the [FAQ](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/docs/faq.md).
- [x] I've read [Instructions for logging issues](https://github.com/badeball/cypress-cucumber-preprocessor/blob/master/CONTRIBUTING.md#instructions-for-logging-issues).
- [x] I'm not using `cypress-cucumber-preprocessor@4.3.1` (package name has changed and it is no longer the most recent version, see [#689](https://github.com/badeball/cypress-cucumber-preprocessor/issues/689)).

**Comment by badeball** [comment]
Looks like this only affect the debug output and that the proper options _are_ actually taking place. Are you experiencing anything different?

**Comment by badeball** [comment]
This has been fixed in v17.0.0.
